environment/environment.go

```diff
@@ -81,20 +86,11 @@ func (env *Environment) Export(ctx context.Context) (rerr error) {
 
 }
 
-func (env *Environment) State(ctx context.Context) ([]byte, error) {
-	containerID, err := env.container.ID(ctx)
-	if err != nil {
-		return nil, err
-	}
+func (env *Environment) container() *dagger.Container {
+	env.mu.RLock()
+	defer env.mu.RUnlock()
```

Comment:
ah shit, i have this exact thing on a branch with a slightly different `return`... let's both try to have a light touch with callsites or else it's gonna be a very annoying merge conflict.